### PR TITLE
test: integration tests for example modules

### PR DESCRIPTION
**Pull request by bavshin-f5**
* Test the whole process of building Rust code with `--add-dynamic-module` and the resulting modules with our Perl test framework
* Fix doctests for nginx-sys and add to CI
* Fix rustdoc warnings and add to CI
* Fix all remaining clippy warnings

### CODE REVIEW

File: .github/workflows/nginx.yaml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        nginx-ref:
+          # master
+          - stable-1.26
+
+    env:
```
Comment (thresheek) [inline .github/workflows/nginx.yaml:36]: Those probably don't need to be defined for steps other than the actual prove run.

File: .github/workflows/nginx.yaml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+          restore-keys: ${{ runner.os }}-cargo-
+
+      - name: Configure nginx
+        working-directory: nginx
+        run: |
+          ${NGX_CONFIGURE} --add-dynamic-module=${{ github.workspace }}/examples
+
+      - name: Build nginx
+        working-directory: nginx
+        run: make
```
Comment (thresheek) [inline .github/workflows/nginx.yaml:74]: Maybe -j something to speed things up

**Review by xeioex — COMMENTED** [review]
Nitpicking: nginx_sys -> nginx-sys?

File: examples/t/upstream.t [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    upstream u {
+        server 127.0.0.1:8081;
+        custom 32;
+    }
+
+    server {
+        listen       127.0.0.1:8080;
+        server_name  localhost;
+
+        error_log %%TESTDIR%%/e_debug.log debug;
```
Comment (xeioex) [inline examples/t/upstream.t:46]: Can't we test it without looking into debug log?

**Review by xeioex — APPROVED** [review]
Otherwise looks good.

File: examples/t/upstream.t (hunk 3 above)
Comment (bavshin-f5) [reply examples/t/upstream.t:46]: This module has no other testable behavior. And I wanted to see if the logs are working before I start breaking these.

**Pull request merged**